### DISCUSSION TITLE
Is it possible to drop multiple commits in a rebase?

### DISCUSSION THREAD

**Discussion by edhgoose** [jesseduffield/lazygit#3005] [Q&A]
Sometimes I decide to rebase a branch and want to drop a large number of commits. Usually keeping 2-3 and otherwise resetting back to the base.

At the moment I usually do it in vim where I can easily delete everything below.

Is there a way in lazygit to say "drop everything below this commit"?

**Comment by stefanhaller** [comment]
If all you want to do is drop a range of commits _at the end_ of your branch, the fastest way to do that (with lazygit or otherwise) is to reset hard to the last commit before that range. In lazygit, select the last commit you want to keep, and press "r", "h".

Dropping a range from the middle of your branch is not as easy; right now, there's no better way to do that then to "e" some commit before the range, then set all commits to "drop" by going "d", down, "d", down repeatedly. A better way to do batch operations like this would be nice (not just for dropping); we are thinking about adding range selection to list views, maybe similar to how you select a range of lines in the staging view.

**Reply by jesseduffield** [reply]
Minor correction: for hard resetting to a commit it's 'g' 'h' rather than 'r' 'h'.
